source/exe/main.cc

```diff
@@ -28,6 +30,7 @@ class ProdComponentFactory : public ComponentFactory {
 } // Server
 
 int main(int argc, char** argv) {
+  ares_library_init(ARES_LIB_INIT_ALL);
```

Comment:
I think ideally this file should be agnostic as to what the underlying implementation of DNS is. FWIW `ares_library_init` is a nop except on windows using winsock. 

It also has some sort of reference counter so you could call `ares_library_init` on the constructor of `DnsResolverImpl` and `ares_library_cleanup` on the destructor. That way the cleanup (if at any point there is any for non windows) will actually happen when the last `DnsResolverImpl` is destroyed.